### PR TITLE
Redis authentication on ICDS

### PR DESCRIPTION
**Pull request by rohit-dimagi**
##### SUMMARY

* Redis authentication on ICDS

##### ENVIRONMENTS AFFECTED

* ICDS
##### ISSUE TYPE

- Change Pull Request

### CODE REVIEW

**Comment by rohit-dimagi** [comment]
in order to support this we need this release https://github.com/DavidWittman/ansible-redis/releases/tag/1.2.7 
i don't see much have changed here but i'd need a second opinion.

File: src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2 [hunk 1]
```diff
@@ -400,7 +400,11 @@ REPORT_CACHE = 'redis'
...
     'BACKEND': 'django_redis.cache.RedisCache',
 {% if 'REDIS_DB' in localsettings %}
+{% if localsettings.get('REDIS_PASSWORD') %}
+    'LOCATION': 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}/{{ localsettings.REDIS_DB }}',
+{% else %}
     'LOCATION': 'redis://{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}/{{ localsettings.REDIS_DB }}',
+{% endif %}
 {% else %}
     'LOCATION': 'redis://{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}',
 {% endif %}
```
Comment (snopoke) [inline src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2]: How about this change so we can support password even if you don't specify a database? Also it prevents having nested if's

```suggestion
{% set redis_url = 'redis://:{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
{% if localsettings.get('REDIS_PASSWORD') %}
{% set redis_url = 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
{% endif %}
{% if 'REDIS_DB' in localsettings %}
    'LOCATION': '{{ redis_url }}/{{ localsettings.REDIS_DB }}',
{% else %}
    'LOCATION': '{{ redis_url }}',
{% endif %}
```

**Comment by snopoke** [comment]
> in order to support this we need this release https://github.com/DavidWittman/ansible-redis/releases/tag/1.2.7
i don't see much have changed here but i'd need a second opinion.

Looks ok to me

File: src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2 [hunk 2]
```diff
@@ -399,10 +399,14 @@ REPORT_CACHE = 'redis'
 
 redis_cache = {
     'BACKEND': 'django_redis.cache.RedisCache',
+{% set redis_url = 'redis://:{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
```
Comment (snopoke) [inline src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2]:
```suggestion
{% set redis_url = 'redis://:{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}' %}
```

File: src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2 [hunk 3]
```diff
@@ -399,10 +399,14 @@ REPORT_CACHE = 'redis'
...
+{% if localsettings.get('REDIS_PASSWORD') %}
+{% set redis_url = 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}'
```
Comment (snopoke) [inline src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2]:
```suggestion
{% set redis_url = 'redis://:{{ localsettings.REDIS_PASSWORD}}@{{ localsettings.REDIS_HOST }}:{{ localsettings.REDIS_PORT }}' %}
```

**Pull request merged**